### PR TITLE
Support k8s 1.22

### PR DESCRIPTION
**Pull request by nijave**
Took a crack at updating API versions to support k8s v1.22

I'm guessing this simultaneously removes support for older versions of k8s but not sure how that relates to usage of this code/if that's an issue

I don't do much go development and haven't worked with k8s APIs so don't have much context on how/what this stuff does

### CODE REVIEW

File: cmd/manager/main.go [hunk 1]
```diff
@@ -29,9 +29,12 @@ func main() {
 
 	printVersion()
 
-	namespace, err := k8sutil.GetWatchNamespace()
-	if err != nil {
-		klog.Fatalf("failed to get watch namespace: %v", err)
+	// see https://github.com/operator-framework/operator-sdk/issues/3948#issuecomment-701524174
+	var namespace string
+	namespace, hasErr := os.LookupEnv("WATCH_NAMESPACE")
```
Comment (nijave) [inline cmd/manager/main.go:31]: https://github.com/operator-framework/operator-sdk/issues/3948#issuecomment-701524174

File: cmd/manager/main.go [hunk 2]
```diff
@@ -2,23 +2,21 @@ package main
...
 	"sigs.k8s.io/controller-runtime/pkg/client/config"
 	"sigs.k8s.io/controller-runtime/pkg/manager"
-	"sigs.k8s.io/controller-runtime/pkg/runtime/signals"
+	"sigs.k8s.io/controller-runtime/pkg/manager/signals"
 )
 
 func printVersion() {
 	klog.V(2).Infof("Go Version: %s", runtime.Version())
 	klog.V(2).Infof("Go OS/Arch: %s/%s", runtime.GOOS, runtime.GOARCH)
-	klog.V(2).Infof("operator-sdk Version: %v", sdkVersion.Version)
```
Comment (nijave) [inline cmd/manager/main.go:21]: This info doesn't appear to be exposed anywhere, that I could tell

See https://github.com/operator-framework/operator-sdk/commits/master/internal/version/version.go (it's been progressively getting removed)

**Comment by sdlarsen** [comment]
I'd love to see a release with 1.22 support...

**Comment by sdlarsen** [comment]
I tried this out on a 1.22.1 cluster. I had to modify the appendApprovalCondition function, setting the Status field to true. Other than that, it seems to work fine. Thank you for the effort.

```
func appendApprovalCondition(csr *capi.CertificateSigningRequest, message string) {
	csr.Status.Conditions = append(csr.Status.Conditions, capi.CertificateSigningRequestCondition{
		Type:    capi.CertificateApproved,
		Status:  "True",
		Reason:  "AutoApproved by kubelet-rubber-stamp",
		Message: message,
	})
}
```

**Comment by nijave** [comment]
> I tried this out on a 1.22.1 cluster. I had to modify the appendApprovalCondition function, setting the Status field to true. Other than that, it seems to work fine. Thank you for the effort.
> 
> ```
> func appendApprovalCondition(csr *capi.CertificateSigningRequest, message string) {
> 	csr.Status.Conditions = append(csr.Status.Conditions, capi.CertificateSigningRequestCondition{
> 		Type:    capi.CertificateApproved,
> 		Status:  "True",
> 		Reason:  "AutoApproved by kubelet-rubber-stamp",
> 		Message: message,
> 	})
> }
> ```

What happened before you changed that? Any errors/etc?

Seems like that field should be optional based on the docs but maybe I'm just reading them wrong

**Comment by sdlarsen** [comment]
I had an error telling me the Status was missing, which was why I added it. I did find the docs a little unclear, but after reading it a few times I think it's actually mandatory. In any case it works with Status set and not without, which is an indication too. I've had it running for quite a while in a 1.22 cluster and it keeps working, so I think it's solid.

**Comment by invidian** [comment]
I can confirm that this patch is working as expected for me.

**Comment by lbogdan** [comment]
Until this PR gets merged, can someone maybe publish an image built from it?

**Comment by invidian** [comment]
I maintain `quay.io/flexkube/kubelet-rubber-stamp-amd64:0.3.0-k8s-1.22`.

**Pull request closed without merging**